Add GitHub issue forms (#222)
trinodb/trino-python-client: .github/ISSUE_TEMPLATE/feature_request.md

---
name: Feature request
about: Suggest an idea for Trino Python client
title: ''
labels: enhancement
assignees: ''

---

### Describe the feature

> A clear and concise description of what you want to happen.

Reviewer: How about asking for a problem statement first, and then the feature description that should explain how it addresses the problem?

### Describe alternatives you've considered
A clear and concise description of any alternative solutions or features you've considered.

> ### Additional context

Reviewer 2: Very vague and non-actionable.

Author: I will remove this section.

Please include any other relevant context here.

> ### Who will this benefit?

Reviewer: Obviously the author :-) I think it's too vague, a problem statement and a solution should be sufficient.

Author: I think this may let the author think more about which persona will benefit from the feature and leave less to the imagination of the person producing the PR.

Reviewer 2: But what would you use that information for?

Author: I agree it does not necessarily need a separate section but knowing who the feature is for, is part of a good problem statement. It will help people detect gaps in the feature or may trigger more reflection.

> What kind of use case will this feature be useful for? Please be specific and provide examples, this will help us prioritize properly.

Reviewer 2: The Trino python client is an open source project and there is no JIRA or task management involved - so there are no priorities. Things will get done when they'll get done and depending on how much effort someone puts in for a certain thing to get done.

cc: @martint

Reviewer 3: While there may be no formal project management in place, there is some prioritization happening.  A bug leading to incorrect results will get more attention than a cosmetic issue.  Same thing with a feature that has a wider audience vs. a narrow use case.  An issue may be "prioritized" by virtue of getting more contributors interested in implementing it, because of its potential impact.

Reviewer 4: Every project, including open source has priorities, bugs vs new features at least.
I don't see any issues with this sentence.

Reviewer: This is about managing expectations. I'd avoid suggesting that any reported issues will have any priority. For non-native English speakers, this might be confusing, even to the point they may think that all issues have _some_ priority, where usually all of them are handled with best effort.

tl;dr let's avoid building up expectations we can't meet.

> ### Are you interested in contributing this feature?

Reviewer: Maybe to make the template a bit shorter we should just ask users to assign themselves to the issue. All Trino contributors that signed the CLA are added to the `contributors` group and should be able to do that. If not, just leave a comment.

Reviewer 2: Yes, just ask people to assign themselves if they're already working on a PR - helps other people who see the issue know to not duplicate work as well.

Author: I think why a lot of other open source projects explicitly ask for this is that it increases the influx of new contributors and shows that the community is there to help. This is probably not so useful for a long time Trino contributor. See eg https://github.com/apache/airflow/issues/new?assignees=&labels=kind%3Abug%2Carea%3Acore&template=airflow_bug_report.yml

Reviewer 2: Good point. Let's keep this and we can assign (or tell them to assign themselves). Let's iterate. Doesn't need to be perfect on first go.

---
name: Feature request
about: Suggest an idea for Trino Python client
title: ''
labels: enhancement
assignees: ''

---

### Describe the feature

> <!-- A clear and concise description of what you want to happen.

Reviewer: This is still missing a "why":
```suggestion
<!-- A clear and concise description of what you want to happen and what problem it would solve. 
```

> <!-- A clear and concise description of what you want to happen.
> What kind of use case will this feature be useful for?

Reviewer: I'm afraid when asked for a use case most people would just repeat the solution instead of defining the problem. We should ask for it explicitly.
```suggestion
<!-- A clear and concise description of what you want to happen
and what problem it would solve.
```
